### PR TITLE
Add: introduce splitjoin

### PR DESCRIPTION
**Pull request by gbprod**
### PR Prelude

- [x] I have read and understood SpaceVim's [CONTRIBUTING](https://github.com/SpaceVim/SpaceVim/blob/master/CONTRIBUTING.md) document.
- [x] I have read and understood SpaceVim's [CODE_OF_CONDUCT](https://github.com/SpaceVim/SpaceVim/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I understand my PR may be closed if it becomes obvious I didn't actually perform all of these steps.

### Why this change is necessary and useful?

Introduce [splitjoin](https://github.com/AndrewRadev/splitjoin.vim) plugin in edit layer.

### CODE REVIEW

**Comment by gbprod** [comment]
I'm not shure about the key bindings, I would like to add it to the "SPC menu", something like `SPC x j i` or `SPC J i` but I can't find a available and mnemonic one...

Also, I think it could be relevant to replace jplus plugin by the more complete [Join](https://github.com/sk1418/join), what do you think ?

**Comment by wsdjeg** [comment]
yeah, I will check these two plugins, and we should not use `ctrl-j` key binding, becaseu this key binding is used in tmux layer, and some people also use it to jump to windows below current one. and maybe we can use `SPC j`, and the description of `SPC j` is `+Jump/Join/Split` now.

**Comment by gbprod** [comment]
There's also the awesome splitjoin plugin https://github.com/AndrewRadev/splitjoin.vim

I will try to improve this PR !

**Comment by gbprod** [comment]
I've updated to introduce splitjoin plugin as a layer.

I would like to add the "Join with delimiter" binding but maybe in another PR

**Comment by wsdjeg** [comment]
`jplus` has been included in edit layer, how about add this plugin into edit layer, layer should be a group of plugins which provide relative functions,

File: autoload/SpaceVim/layers/edit.vim [hunk 1]
```diff
@@ -32,6 +32,7 @@ function! SpaceVim#layers#edit#plugins() abort
         \ [g:_spacevim_root_dir . 'bundle/vim-jplus', { 'on_map' : '<Plug>(jplus' }],
         \ [g:_spacevim_root_dir . 'bundle/tabular',           { 'merged' : 0}],
         \ [g:_spacevim_root_dir . 'bundle/vim-better-whitespace',  { 'on_cmd' : ['StripWhitespace', 'ToggleWhitespace', 'DisableWhitespace', 'EnableWhitespace']}],
+        \ ['andrewradev/splitjoin.vim',{ 'merged' : 0, 'loadconf' : 1}],
```
Comment (gbprod) [inline autoload/SpaceVim/layers/edit.vim:35]: Maybe I should add this plugin to "bundle" ?

Comment (wsdjeg) [reply autoload/SpaceVim/layers/edit.vim:35]: Yeah, because this layer is enabled by default, all plugins in this layer should be included in bundle dir.

**Pull request merged**